internal/apiserver/root.go

```diff
@@ -498,6 +505,54 @@ func setNamespaceOperatingMode() error {
 	return nil
 }
 
+// setRandomPgouserPasswords looks through the pgouser secrets in the Operator's
+// namespace. If any have an empty password, it generates a random password,
+// Base64 encodes it, then stores it in the relevant PGO user's secret
+func setRandomPgouserPasswords() {
+	ctx := context.TODO()
+
+	selector := "pgo-pgouser=true,vendor=crunchydata"
+	secrets, err := Clientset.CoreV1().Secrets(PgoNamespace).
+		List(ctx, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		log.Warnf("Could not get pgouser secrets in namespace: %s", PgoNamespace)
+		return
+	}
+
+	for _, secret := range secrets.Items {
+		// check if password is empty
+		if len(secret.Data["password"]) > 0 {
+			continue
+		}
+
+		log.Infof("Password in pgouser secret %s for operator installation %s in namespace %s is empty. "+
+			"Setting a generated password.", secret.Name, InstallationName, PgoNamespace)
+
+		// generate the password using the default password length
+		generatedPassword, err := util.GeneratePassword(util.DefaultGeneratedPasswordLength)
+
+		if err != nil {
+			log.Errorf("Could not generate password for pgouser secret %s for operator installation %s in "+
+				"namespace %s", secret.Name, InstallationName, PgoNamespace)
+		}
+
+		// create the password patch
+		patch, err := kubeapi.NewMergePatch().Add("stringData", "password")(generatedPassword).Bytes()
+
+		if err != nil {
+			log.Errorf("Could not generate password patch for pgouser secret %s for operator installation "+
+				"%s in namespace %s", secret.Name, InstallationName, PgoNamespace)
+		}
```

Comment:
These continue to process after logging. Perhaps `continue` when there's an error.

Reply:
Agreed, updated.